src/Core/Webservice/WebserviceCanBeEnabledConfigurationChecker.php

```diff
@@ -0,0 +1,171 @@
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Webservice;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Translation\TranslatorInterface;
+use RuntimeException;
+
+/**
+ * Looks at server configuration in order to check PrestaShop Webservice can be enabled
+ */
+class WebserviceCanBeEnabledConfigurationChecker
+{
+    const ISSUE_NOT_APACHE_SERVER = 'not_apache_server';
+    const ISSUE_CANNOT_CHECK_APACHE_MODULES = 'cannot_check_apache_modules';
+    const ISSUE_APACHE_MOD_AUTH_BASIC_NOT_AVAILABLE = 'issue_apache_mod_auth_basic_not_available';
+    const ISSUE_APACHE_MOD_AUTH_REWRITE_NOT_AVAILABLE = 'issue_apache_mod_auth_rewrite_not_available';
+    const ISSUE_EXT_SIMPLEXML_NOT_AVAILABLE = 'issue_ext_simplexml_not_available';
+    const ISSUE_HTTPS_NOT_AVAILABLE = 'issue_https_not_available';
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param Configuration $configuration
+     */
+    public function __construct(TranslatorInterface $translator, Configuration $configuration)
+    {
+        $this->translator = $translator;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * Analyses the server configuration (apache configuration and php settings)
+     * to check whether PrestaShop Webservice can be used or not.
+     *
+     * @param Request $request optional
+     *
+     * @return array empty if no errors
+     */
+    public function getErrors(Request $request = null)
+    {
+        $issues = $this->lookForIssues($request);
+
+        $allWarningMessages = $this->getWarningMessages();
+        $selectedWarningMessages = [];
+
+        foreach ($issues as $issue) {
+            if (false === array_key_exists($issue, $allWarningMessages)) {
+                throw new RuntimeException(sprintf('Unexpected configuration issue %s', $issue));
```

Comment:
Watch out with RuntimeException you never catch it, can produce errors because of hooks :/

Reply:
I want this RuntimeException to never be catched because if it is thrown it indicates that someone modified `WebserviceCanBeEnabledConfigurationChecker` and made a mistake in doing so. This RuntimeException is for her/him :) so he can detect easily his mistake.